store/sqlstore/store.go

```diff
@@ -4,6 +4,1346 @@
+	EXIT_ALTER_COLUMN                = 116
+	EXIT_CREATE_INDEX_POSTGRES       = 117
+	EXIT_CREATE_INDEX_MYSQL          = 118
+	EXIT_CREATE_INDEX_FULL_MYSQL     = 119
+	EXIT_CREATE_INDEX_MISSING        = 120
+	EXIT_REMOVE_INDEX_POSTGRES       = 121
+	EXIT_REMOVE_INDEX_MYSQL          = 122
+	EXIT_REMOVE_INDEX_MISSING        = 123
+	EXIT_REMOVE_TABLE                = 134
+	EXIT_CREATE_INDEX_SQLITE         = 135
+	EXIT_REMOVE_INDEX_SQLITE         = 136
+	EXIT_TABLE_EXISTS_SQLITE         = 137
+	EXIT_DOES_COLUMN_EXISTS_SQLITE   = 138
+	EXIT_ALTER_PRIMARY_KEY           = 139
+)
+
+type SqlStoreStores struct {
+	team                 store.TeamStore
+	channel              store.ChannelStore
+	post                 store.PostStore
+	thread               store.ThreadStore
+	user                 store.UserStore
+	bot                  store.BotStore
+	audit                store.AuditStore
+	cluster              store.ClusterDiscoveryStore
+	compliance           store.ComplianceStore
+	session              store.SessionStore
+	oauth                store.OAuthStore
+	system               store.SystemStore
+	webhook              store.WebhookStore
+	command              store.CommandStore
+	commandWebhook       store.CommandWebhookStore
+	preference           store.PreferenceStore
+	license              store.LicenseStore
+	token                store.TokenStore
+	emoji                store.EmojiStore
+	status               store.StatusStore
+	fileInfo             store.FileInfoStore
+	uploadSession        store.UploadSessionStore
+	reaction             store.ReactionStore
+	job                  store.JobStore
+	userAccessToken      store.UserAccessTokenStore
+	plugin               store.PluginStore
+	channelMemberHistory store.ChannelMemberHistoryStore
+	role                 store.RoleStore
+	scheme               store.SchemeStore
+	TermsOfService       store.TermsOfServiceStore
+	productNotices       store.ProductNoticesStore
+	group                store.GroupStore
+	UserTermsOfService   store.UserTermsOfServiceStore
+	linkMetadata         store.LinkMetadataStore
+}
+
+type SqlStore struct {
+	// rrCounter and srCounter should be kept first.
+	// See https://github.com/mattermost/mattermost-server/v5/pull/7281
+	rrCounter      int64
+	srCounter      int64
+	master         *gorp.DbMap
+	replicas       []*gorp.DbMap
+	searchReplicas []*gorp.DbMap
+	stores         SqlStoreStores
+	settings       *model.SqlSettings
+	lockedToMaster bool
+	context        context.Context
+	license        *model.License
+	licenseMutex   sync.RWMutex
+}
+
+type TraceOnAdapter struct{}
+
+func (t *TraceOnAdapter) Printf(format string, v ...interface{}) {
+	originalString := fmt.Sprintf(format, v...)
+	newString := strings.ReplaceAll(originalString, "\n", " ")
+	newString = strings.ReplaceAll(newString, "\t", " ")
+	newString = strings.ReplaceAll(newString, "\"", "")
+	mlog.Debug(newString)
+}
+
+func NewSqlStore(settings model.SqlSettings, metrics einterfaces.MetricsInterface) *SqlStore {
```

Comment:
While we are here, I would suggest to change the name to just `Store` to avoid the stutter in `sqlstore.NewSqlStore`. It should just be `sqlstore.New`, and the struct should be `Store`.

https://golang.org/doc/effective_go.html#package-names

> The importer of a package will use the name to refer to its contents, so exported names in the package can use that fact to avoid stutter.

> For instance, the buffered reader type in the bufio package is called Reader, not BufReader, because users see it as bufio.Reader, which is a clear, concise name.

Reply:
I totally agree with the `New` method name, but I don't know if the `sqlstore.Store` is going to be confusing with `store.Store`, for this case I actually prefer the `SqlStore`.

Reply:
Ok.